react/features/filmstrip/components/web/ParticipantName.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import { makeStyles } from '@material-ui/core/styles';
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import { getLocalParticipant, getParticipantDisplayNameWithId } from '../../../base/participants';
+import { withPixelLineHeight } from '../../../base/styles/functions.web';
+import { DisplayNameBadge, appendSuffix } from '../../../display-name';
+
+type Props = {
+
+    /**
+     * The participant ID.
+     */
+    participantId: string,
+
+    /**
+     * The suffix to be appended to the participant name.
+     */
+    participantSuffix?: string
+}
+
+const useStyles = makeStyles(theme => {
+    return {
+        badgeContainer: {
+            ...withPixelLineHeight(theme.typography.labelRegular),
+            alignItems: 'center',
+            bottom: theme.spacing(2),
+            display: 'flex',
+            justifyContent: 'center',
+            position: 'absolute',
+            transition: 'margin-bottom 0.3s',
+            width: '100%',
+            zIndex: '2'
+        }
+    };
+});
+
+/**
+ * Component that renders a badge with the participant name on each thumbnail in tile view.
+ *
+ * @returns {ReactElement}
+ */
+const ParticipantName = ({ participantId, participantSuffix }: Props) => {
+    const classes = useStyles();
+    const participantName = useSelector(getParticipantDisplayNameWithId(participantId));
```

Comment:
I'm wondering is there a performance impact from passing an anonymous function as a selector?

Reply:
There shouldn't be any performance issues as the anonymous function is only imported once and it is never redefined, useSelector takes care of the memoization AFAIK.

Reply:
Hmm. Isn't `getParticipantDisplayNameWithId` non-anonymous?

Actually what I'm worried about is that `getParticipantDisplayNameWithId(participantId)` is returning an anonymous function. Is the memoization going to work in this case, since we will actually provide different function to `useSelector` every time the component is re-rendered?